src/gui/titlebar.cpp

```diff
@@ -324,10 +324,17 @@ static void maybe_add_recording_pause_mark(std::string& title_str)
 
 static void set_window_title()
 {
-	auto title_str = state.title_no_tags;
-	maybe_add_muted_mark(title_str);
-	maybe_add_recording_pause_mark(title_str);
-	SDL_SetWindowTitle(sdl.window, title_str.c_str());
+	static std::string last_title_str = {};
+
+	auto new_title_str = state.title_no_tags;
+
+	maybe_add_muted_mark(new_title_str);
+	maybe_add_recording_pause_mark(new_title_str);
+
+	if (new_title_str != last_title_str) {
+		SDL_SetWindowTitle(sdl.window, new_title_str.c_str());
+		last_title_str = new_title_str;
+	}
```

Comment:
This is probably fine.  I was thinking to wrap the entire `GFX_RefreshTitle()` function like it was before but now that I look at it more, the function looks safe even if it gets called a lot.  The animation code is just stopping/starting a timer when it needs to.

By "a lot", I was worried this would get called every tick if the CPU throttles and changes cycles but you said that is not the case so probably nothing to fear.

Reply:
Yeah I thought it was safe without this too, but this precautionary diffing is just good practice.

I couldn't wrap it like before as the title bar update gets triggered by the cycles changed notification. Well, I could restrict the diffing to the human readable cycles text only and not call `GFX_RefreshTile()` if no change, but I like this generalised solution better to diff the whole title bar text.